### PR TITLE
[TH2-365] Corrected README

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
* Added example of CR
* Renamed queues to pins

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,44 @@
...
+
+Connect has 4 type of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+
+- incoming parsed messages // it will be removed in future
+- outgoing parsed messages // it will be removed in future
 - incoming raw messages
 - outgoing raw messages
 
-To send messages, a separate queue is used. The Connectivity component subscribes to that queue at the start and waits for the messages - to be sent to the target system - to come from the queue.
+To send messages, a separate queue is used. The Connect component subscribes to that pin at the start and waits for the messages - to be sent to the target system - to come from the pin.
```
Comment (OptimumOpium) [inline README.md:18]:
```suggestion
To send messages, a separate queue is used. The Connect component subscribes to that pin at the start and waits for the messages. The messages received from that pin will be sent to the target system.
```

File: README.md [hunk 2]
```diff
@@ -1,35 +1,44 @@
 ## How it works
 
-The Connectivity component is responsible for the communication with the target system. This component implements the logic of the interaction protocol, receives and sends messages from and to the system, respectively.
+The Connect component is responsible for the communication with the target system. This component implements the logic of the interaction protocol, receives and sends messages from and to the system, respectively.
 
-## Queues
+## Appointment
 
-Connectivity has 4 queues for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+This project include only adapter logic between Sailfish and th2 packed into Docker Image. This image used in [th2-conn-generic](https://github.com/th2-net/th2-conn-generic) as base.
```
Comment (OptimumOpium) [inline README.md:7]:
```suggestion
This project includes only an adapter for using Sailfish in the th2 packed into Docker Image. This image is used in [th2-conn-generic](https://github.com/th2-net/th2-conn-generic) as base.
```

File: README.md [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
 
-- incoming parsed messages
-- outgoing parsed messages
+## Pins
+
+Connect has 4 type of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
```
Comment (OptimumOpium) [inline README.md:11]:
```suggestion
Connect has 4 types of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
```

**Review by OptimumOpium — CHANGES_REQUESTED** [review]

**Review by OptimumOpium — DISMISSED** [review]

**Review by OptimumOpium — APPROVED** [review]
Approved because of self-review

File: README.md [hunk 4]
```diff
@@ -1,35 +1,116 @@
...
+This image should be used as a base to implement extensions with the real logic for specific protocols using services in the Sailfish format.
 
-Connectivity has 4 queues for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+As an example, the [th2-conn-generic](https://github.com/th2-net/th2-conn-generic) project implements the extension for connecting via FIX protocol using standard Sailfish's FIX service.
 
-- incoming parsed messages
-- outgoing parsed messages
+## Configuration
+
+This configuration should be specified in the custom configuration block in schema editor.
```
Comment (jean-michel-perli-sonsin) [inline README.md:13]: This configuration should be specified in the custom configuration block **at the** schema editor.

File: README.md [hunk 5]
```diff
@@ -1,35 +1,116 @@
...
+session-alias: "connectivity-alias"
+workspace: "/home/sailfish/workspace"
+type: "th2_service:Your_Service_Type"
+name: "your_service"
+settings:
+  param1: "value1"
+```
+
+Parameters:
++ session-alias - that session alias will be set for all messages received or sent by this component. **It should be unique for each "Connect" component**;
```
Comment (jean-michel-perli-sonsin) [inline README.md:25]: **the session alias specified in this field,** will be set for all messages received or sent by this component.

File: README.md [hunk 6]
```diff
@@ -1,35 +1,116 @@
...
++ workspace - the folder inside the container that will contain a plugin adapted to use in the TH2;
```
Comment (jean-michel-perli-sonsin) [inline README.md:26]: the folder inside the container **which** will contain a plugin adapted **for using with** TH2

File: README.md [hunk 7]
```diff
@@ -1,35 +1,116 @@
...
++ type - the service type from **services.xml** file. If service name from services.xml file contains `-` symbols they must be replaced with `_` symbol;
```
Comment (jean-michel-perli-sonsin) [inline README.md:27]: the service type **specified** from services.xml file. If **the** service name from services.xml file contains **the symbols `-`**, they must be replaced with **the symbol `_`**;

File: README.md [hunk 8]
```diff
@@ -1,35 +1,116 @@
...
++ name - the service name that will be displayed in the events in the report;
```
Comment (jean-michel-perli-sonsin) [inline README.md:28]: the service name **which** will be displayed **into the events inside the report**;

File: README.md [hunk 9]
```diff
@@ -1,35 +1,116 @@
...
++ settings - the parameters that will be transformed to the actual service's settings specified in the **services.xml** file.
+
+## Extension
+
+You can add the ability to connect to a target system by implementing your own service in the Sailfish format and putting it and its configuration to the correct places in the base image.
```
Comment (jean-michel-perli-sonsin) [inline README.md:33]: You can add the ability to connect to a target system by implementing your own service in the Sailfish format and **by** putting it **together with its** configuration to the correct places **into** the base image.

File: README.md [hunk 10]
```diff
@@ -1,35 +1,116 @@
...
+
+You need to take the following steps:
```
Comment (jean-michel-perli-sonsin) [inline README.md:35]: You need to **perform** the following steps:

File: README.md [hunk 11]
```diff
@@ -1,35 +1,116 @@
...
+
+1. Create the implementation of the [com.exactpro.sf.services.IService](https://github.com/exactpro/sailfish-core/blob/master/BackEnd/Core/sailfish-core/src/main/java/com/exactpro/sf/services/IService.java).
```
Comment (jean-michel-perli-sonsin) [inline README.md:37]: Create the implementation of

File: README.md [hunk 12]
```diff
@@ -1,35 +1,116 @@
...
+The examples of implementing this interface can be found [here](https://github.com/exactpro/sailfish-core/tree/master/BackEnd/Service).
+If the protocol is already implemented in the Sailfish services you can simply use the dependencies on the service's artifact for that protocol.
+
+2. Create the **services.xml** configuration file that contains the description for services you can use from that "Connect" component.
```
Comment (jean-michel-perli-sonsin) [inline README.md:41]: Create the services.xml configuration file **which** contains the description for **the** services **that** you can use from **the** "Connect" component.

File: README.md [hunk 13]
```diff
@@ -1,35 +1,116 @@
...
+    plugin_alias: th2_service
+    name: th2_service
+    build_number: 0
+    revision: 0
+    git_hash: 0
+    branch: fake
+    version: 3.2.0.0
+    core_version: 3.2.0
+    ```
+4. Create you own image based on the current one and put all files to the correct places in the base image:
```
Comment (jean-michel-perli-sonsin) [inline README.md:61]: Create you own image based on the current one and put all **the** files to the correct places **into** the base image:

File: README.md [hunk 14]
```diff
@@ -1,35 +1,116 @@
...
+    + Create the following directory - **${workspace}/plugins/th2_service**.
+    _**${workspace}**_ - it is a folder from the "Connect" configuration.
```
Comment (jean-michel-perli-sonsin) [inline README.md:63]: **is a** folder from the "Connect" configuration.

File: README.md [hunk 15]
```diff
@@ -1,35 +1,116 @@
...
+    If you use the _plugin_alias_ and _name_ different from _th2_service_ in the VERSION file correct the _th2_service_ folder name according to value you use.
```
Comment (jean-michel-perli-sonsin) [inline README.md:64]: If you use the _plugin_alias_ and _name_ different from _th2_service_ in the VERSION file correct the _th2_service_ folder name according **to the value that you use.**

File: README.md [hunk 16]
```diff
@@ -1,35 +1,116 @@
...
+    Let's name that directory as **PLUGIN_DIRECTORY** for simplicity. This name will be used in future steps.
+    + Artifact with the service(s) implementation(s) and all its dependencies should be put to the following directory - **${PLUGIN_DIRECTORY}/libs**.
```
Comment (jean-michel-perli-sonsin) [inline README.md:66]: Artifact with the service(s) implementation(s) and all its dependencies should be **placed into** the following directory - ${PLUGIN_DIRECTORY}/libs.

File: README.md [hunk 17]
```diff
@@ -1,35 +1,116 @@
...
+    + The configuration file created on the step 2 should be put to the following directory - **${PLUGIN_DIRECTORY}/cfg**.
```
Comment (jean-michel-perli-sonsin) [inline README.md:67]: The configuration file created on the step 2 should be **placed into** the following directory - ${PLUGIN_DIRECTORY}/cfg.

File: README.md [hunk 18]
```diff
@@ -1,35 +1,116 @@
...
+    + The _VERSION_ file created on step 3 should be put to the following directory - **${PLUGIN_DIRECTORY}/**.
```
Comment (jean-michel-perli-sonsin) [inline README.md:68]: The _VERSION_ file created on step 3 should be **placed into** the following directory - ${PLUGIN_DIRECTORY}/.

File: README.md [hunk 19]
```diff
@@ -1,35 +1,116 @@
...
+
+## Pins
+
+Connect has 4 types of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
```
Comment (jean-michel-perli-sonsin) [inline README.md:72]: Connect has 4 types of pins for interacting with **the** th2 components. Messages that were received from / sent to the target system will be **directed** to the following queues:

File: README.md [hunk 20]
```diff
@@ -1,35 +1,116 @@
...
-- TH2_EVENT_STORAGE_GRPC_HOST=some-host-name-or-ip
-- TH2_EVENT_STORAGE_GRPC_PORT=some-port
-- IN_QUEUE_NAME=in_queue
-- IN_RAW_QUEUE_NAME=in_raw_queue
-- OUT_QUEUE_NAME=out_queue
-- OUT_RAW_QUEUE_NAME=out_raw_queue
-- TO_SEND_QUEUE_NAME=to_send_queue
-- TO_SEND_RAW_QUEUE_NAME=to_send_queue
\ No newline at end of file
+The "Connect" component uses a separate queue to send messages. The component subscribes to that pin at the start and waits for the messages.
```
Comment (jean-michel-perli-sonsin) [inline README.md:79]: The "Connect" component uses a separate queue to send messages. The component subscribes to that pin at the start and **wait for messages to be received**.

File: README.md [hunk 21]
```diff
@@ -1,35 +1,116 @@
...
+The messages received from that pin will be sent to the target system.
```
Comment (jean-michel-perli-sonsin) [inline README.md:80]: **The received messages** from that pin will be sent to the target system.

File: README.md [hunk 22]
```diff
@@ -1,35 +1,116 @@
...
+Also, this component is responsible for maintaining connections and sessions in the cases where this is provided by the communication protocol.
```
Comment (jean-michel-perli-sonsin) [inline README.md:81]: **Besides**, this component is responsible for maintaining connections and sessions **,** **in the cases provided** by the communication protocol.

File: README.md [hunk 23]
```diff
@@ -1,35 +1,116 @@
...
+Here you can automatically send heartbeat messages, send a logon/logout, requests to retransmit messages in the event of a gap, etc.
```
Comment (jean-michel-perli-sonsin) [inline README.md:82]: Here you can automatically send heartbeat messages, **logon/logout, requests** to retransmit messages in the event of a gap, etc.

File: README.md [hunk 24]
```diff
@@ -1,35 +1,116 @@
...
+## Pins
+
+Connect has 4 types of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+
+- incoming parsed messages // it will be removed in future
+- outgoing parsed messages // it will be removed in future
 - incoming raw messages
 - outgoing raw messages
 
-To send messages, a separate queue is used. The Connectivity component subscribes to that queue at the start and waits for the messages - to be sent to the target system - to come from the queue.
```
Comment (jean-michel-perli-sonsin) [inline README.md:14]: **A separate queue is used to send messages.** The Connectivity component subscribes to that queue at the start and waits for the messages **(that will be sent to the target system)** to **be received** from the queue.

File: README.md [hunk 25]
```diff
@@ -1,35 +1,116 @@
...
-Also, this component is responsible for maintaining connections and sessions in the cases where this is provided by the communication protocol. Here you can automatically send heartbeat messages, send a logon/logout, requests to retransmit messages in the event of a gap, etc.
```
Comment (jean-michel-perli-sonsin) [inline README.md:15]: Also, this component is responsible for maintaining connections and sessions in the cases **provided by the communication protocol.** Here you can automatically send heartbeat messages **, logon/logout, requests to** retransmit messages in the event of a gap, etc.

File: README.md [hunk 26]
```diff
@@ -1,35 +1,116 @@
-## How it works
+# Connect
 
-The Connectivity component is responsible for the communication with the target system. This component implements the logic of the interaction protocol, receives and sends messages from and to the system, respectively.
```
Comment (jean-michel-perli-sonsin) [inline README.md:3]: This component **is responsible for implementing the logic of the interaction protocol and for sending and receiving messages from/to the system respectively.**

**Pull request merged**